<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>787</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Mosaic Wall</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QFrame" name="frame">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>12</number>
           </property>
           <property name="topMargin">
            <number>12</number>
           </property>
           <property name="rightMargin">
            <number>12</number>
           </property>
           <item alignment="Qt::AlignHCenter">
            <widget class="ProgressButton" name="btnLoad">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string> Load Base Image</string>
             </property>
             <property name="icon">
              <iconset resource="assets.qrc">
               <normaloff>:/icons/assets/base_image.png</normaloff>:/icons/assets/base_image.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item alignment="Qt::AlignHCenter">
            <widget class="ProgressButton" name="btnSetImageFolder">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string> Set Image Folder</string>
             </property>
             <property name="icon">
              <iconset resource="assets.qrc">
               <normaloff>:/icons/assets/images_folder.png</normaloff>:/icons/assets/images_folder.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lblStatus">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>80</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="btnCancel">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string> Cancel</string>
             </property>
             <property name="icon">
              <iconset resource="assets.qrc">
               <normaloff>:/icons/assets/cancel.png</normaloff>:/icons/assets/cancel.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lblSettings">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Variation</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout_3">
             <property name="spacing">
              <number>10</number>
             </property>
             <item row="1" column="0" colspan="2">
              <widget class="QLabel" name="lblVariation">
               <property name="text">
                <string>disabled</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="0" column="0" colspan="2">
              <widget class="QSlider" name="slHistory">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>5</number>
               </property>
               <property name="singleStep">
                <number>1</number>
               </property>
               <property name="pageStep">
                <number>1</number>
               </property>
               <property name="sliderPosition">
                <number>0</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="invertedControls">
                <bool>false</bool>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksAbove</enum>
               </property>
               <property name="tickInterval">
                <number>1</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="lblOutputResolution">
             <property name="toolTip">
              <string>Resolution of the output image</string>
             </property>
             <property name="text">
              <string>Output Resolution</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1,1" columnminimumwidth="0,0,0">
             <property name="spacing">
              <number>2</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_7">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>50</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Width</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="sbWidth">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="wrapping">
                <bool>false</bool>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="suffix">
                <string/>
               </property>
               <property name="prefix">
                <string/>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>32768</number>
               </property>
               <property name="value">
                <number>3872</number>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QLabel" name="label_8">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>50</width>
                 <height>0</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Height</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QSpinBox" name="sbHeight">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="suffix">
                <string/>
               </property>
               <property name="maximum">
                <number>32768</number>
               </property>
               <property name="value">
                <number>2592</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="btnLockRatio">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="assets.qrc">
                 <normaloff>:/icons/assets/locked.png</normaloff>:/icons/assets/locked.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>16</height>
                </size>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
               <property name="default">
                <bool>false</bool>
               </property>
               <property name="flat">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5"/>
           </item>
           <item alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="btnSetResolution">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string> Set Resolution</string>
             </property>
             <property name="icon">
              <iconset resource="assets.qrc">
               <normaloff>:/icons/assets/resolution.png</normaloff>:/icons/assets/resolution.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="lblGrid">
             <property name="toolTip">
              <string>Defines how ofte the image is devided in rows and columns</string>
             </property>
             <property name="text">
              <string>Grid </string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0" alignment="Qt::AlignHCenter">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Rows</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="sbRows">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>500</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="sbCols">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::UpDownArrows</enum>
               </property>
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>500</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item row="0" column="1" alignment="Qt::AlignHCenter">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Columns</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="btnGenerate">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string> Create Mosaic</string>
             </property>
             <property name="icon">
              <iconset resource="assets.qrc">
               <normaloff>:/icons/assets/generate.png</normaloff>:/icons/assets/generate.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="btnSave">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string> Save</string>
             </property>
             <property name="icon">
              <iconset resource="assets.qrc">
               <normaloff>:/icons/assets/download.png</normaloff>:/icons/assets/download.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="wImageContainer" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ProgressButton</class>
   <extends>QPushButton</extends>
   <header>progressbutton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="assets.qrc"/>
 </resources>
 <connections/>
</ui>
